packages/headless/src/app/headless-engine.ts

```diff
@@ -128,7 +142,8 @@ export interface HeadlessConfigurationOptions {
      */
     searchHub?: string;
     /**
-     * Allows for augmenting request before any (search, facet-search, query-suggest) a request is sent.
+     * Allows for augmenting a request (search, facet-search, query-suggest, etc.) before it is sent.
+     * @deprecated Use `preprocessRequest` instead.
```

Comment:
I would maybe add a middleware for the payload exclusively here, like what the analyticsClientMiddleware does

Reply:
Apart from just the documentation, I would add a log message about the deprecation when the old middleware is ran.

With a precise explanation on how to use the new thing, and that explain that the feature will be removed in upcoming major version.

Reply:
It a dedicated middleware for the payload needed? Or could one use the request modifier to change the payload?

I feel we already have too many middlewares and I do not think it's good to add more unless there is a strong reason to?

Reply:
Those middlewares would be for different use cases, it would look like:
- `options.preprocessRequest`: augment the final request before it is sent, useful for proxies, changing headers, method, url, etc. Passes the request object and the origin of the client, searchAPI, analyticsAPI (fetch and beacon)
- `options.analytics.analyticsClientMiddleware`: modify a payload according to the event type (search, click, custom, view, collect)
- `options.search.searchClientMiddleware` (doesn't exist ATM): modify a payload according to the request type (plan, search, querySuggest, facetSearch, etc.)

Reply:
Why is a `searchClientMiddleware` needed if `preprocessRequest` includes the full request which contains the body payload?

Reply:
It's not absolutely necessary, just like the analyticsClientMiddleware is not.
But it would be more simple for users, separating concern, fulfill use-cases which are a bit different, having access to the payload object before it is stringified to JSON.